specifyweb/businessrules/migrations/0001_initial.py

```diff
@@ -0,0 +1,44 @@
+# Generated by Django 3.2.15 on 2023-09-30 17:35
+
+from django.db import migrations, models
+import django.db.models.deletion
+
+
+class Migration(migrations.Migration):
+
+    initial = True
+
+    dependencies = [
+        ('specify', '__first__'),
+    ]
+
+    operations = [
+        migrations.CreateModel(
+            name='UniquenessRule',
+            fields=[
+                ('id', models.AutoField(auto_created=True, primary_key=True, serialize=False, verbose_name='ID')),
+                ('isdatabaseconstraint', models.BooleanField(default=False)),
+                ('discipline', models.ForeignKey(db_column='DisciplineID', on_delete=django.db.models.deletion.PROTECT, to='specify.discipline')),
+                ('scope', models.ForeignKey(blank=True, db_column='splocalecontaineritemid', null=True, on_delete=django.db.models.deletion.PROTECT, to='specify.splocalecontaineritem')),
+            ],
+            options={
+                'db_table': 'uniquenessrule',
+            },
+        ),
+        migrations.CreateModel(
+            name='UniquenessRule_Splocalecontaineritem',
+            fields=[
+                ('id', models.AutoField(auto_created=True, primary_key=True, serialize=False, verbose_name='ID')),
+                ('splocalecontaineritem', models.ForeignKey(db_column='splocalecontaineritemid', on_delete=django.db.models.deletion.PROTECT, to='specify.splocalecontaineritem')),
+                ('uniquenessrule', models.ForeignKey(on_delete=django.db.models.deletion.CASCADE, to='businessrules.uniquenessrule')),
+            ],
+            options={
+                'db_table': 'uniquenessrule_splocalecontaineritem',
```

Comment:
Both are conventions on the backend: from Specify 6 or prior versions. 

`_` is used in the naming of many-to-many Join tables; there seems to be [two conventions](https://stackoverflow.com/questions/2208186/proper-table-naming-convention-for-a-many-to-many-intersect-table): either something like `Table1Table2` or `Table1_Table2`. The few many-to-many tables currently follow the underscore convention

Since Specify 7's inception, we have not created many-to-many relationships, and if we do interact with them, they are hardcoded operations using raw SQL from a connection and don't properly utilize either Django or SqlAlchemy (See #2928) 

However, there are a few many-to-many join tables which _do_ exist (potentially not an exhaustive list): 
- autonumsch_coll
- autonumsch_div
- autonumsch_dsp
- project_colobj
- specifyuser_spprincipal
- spprincipal_sppermission

Reply:
As discussed before, could you remove the scope field? Rather, you can have a field in the `uniquenessrule_splocalecontaineritem` saying isParent which can be a boolean? This will simplify some of the logic here.

See that this comes more naturally from the new definition of uniqueness rules:
`[["accessionNumber"], ["division"]]`. isParent will be false for accessionNumber and true for division. For fields which are global, just don't have any parent (so set isParent to false). 

It also comes more naturally from how uniqueness rules (constraints are defined) in MySQL database. There is no concept of scope in there too (but we need one, so we can just add isParent here). 

Additonally, this is simpler to change in code(changing one field in one table to change the scope). Also allows multiple scopes (although don't know why that is needed)